### PR TITLE
Validate Data inputs on each save

### PR DESCRIPTION
**Pull request by dblenkus**

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -25,6 +25,7 @@ Changed
 - Parent relations are kept even after the parent is deleted and are deleted
   when the child is deleted
 - Dependency resolver in manager is sped up by use of parent relations
+- Validation of Data inputs is performed on save instead of on create
```
Comment (tjanez) [inline docs/CHANGELOG.rst]: Please, put `Data` in double backticks.

File: resolwe/flow/models/data.py [hunk 2]
```diff
@@ -329,8 +329,11 @@ def save(self, render_name=False, *args, **kwargs):
             if 'update_fields' in kwargs:
                 kwargs['update_fields'].append('size')
 
-        if create:
-            validate_schema(self.input, self.process.input_schema)  # pylint: disable=no-member
+        # Input Data objects are validated only on create as they can be deleted later.
```
Comment (tjanez) [inline resolwe/flow/models/data.py]: "on create" -> "upon creation"

File: resolwe/flow/models/utils.py [hunk 3]
```diff
@@ -77,6 +78,8 @@ def validate_schema(instance, schema, test_required=True, path_prefix=None):
         (default: ``False``)
     :param str path_prefix: path prefix used for checking if files and
         directories exist (default: ``None``)
+    :param bool skip_missing_data: Don't raise an error if referenced
+        Data object does not exist
```
Comment (tjanez) [inline resolwe/flow/models/utils.py]: `Data` -> ``` ``Data`` ```

File: resolwe/flow/tests/test_validation.py [hunk 4]
```diff
@@ -586,6 +586,8 @@ def test_data_field(self):
             self.assertEqual(value_mock.exists.call_count, 1)
             self.assertEqual(value_mock.first.call_count, 0)
 
+            validate_schema(instance, schema, skip_missing_data=True)
```
Comment (tjanez) [inline resolwe/flow/tests/test_validation.py]: 🙂 He he, how you squeezed this one in...

File: docs/CHANGELOG.rst [hunk 5]
```diff
@@ -51,6 +52,7 @@ Fixed
 - Make manager more robust to unhandled failures during data object processing
 - Fix manager deadlock during tests
 - Fix ctypes cache clear during tests
+- Don7t raise ChannelFull error in communicate call
```
Comment (kostko) [inline docs/CHANGELOG.rst]: Replace `7` with `'` ;-)

**Pull request merged**